### PR TITLE
Db obelisk routes encoders

### PR DESCRIPTION
**Pull request by danbornside**

### CODE REVIEW

File: backend-db/Rhyolite/Backend/Schema.hs [hunk 1]
```diff
@@ -100,4 +107,20 @@ instance (PrimitivePersistField (DefaultKey a), DefaultKeyId a) => PrimitivePers
...
+  where
+    pa :: Proxy a = Proxy
+    univ :: [a] = universe
+    failUniv :: Conversion a = conversionError $ VisibleUniverseFailure (typeRep pa)
+    univMap = flip Map.lookup (Map.fromList $ (T.pack . show &&& id) <$> univ)
+
+toShowUniverse :: forall a. Show a => a -> Action
+toShowUniverse = toField . T.pack . show
+
+instance Exception VisibleUniverseFailure
```
Comment (3noch) [inline backend-db/Rhyolite/Backend/Schema.hs]: Should be up next to definition IMO.

File: frontend/Rhyolite/Frontend/Form.hs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  -> m (InputElement EventResult (DomBuilderSpace m) t) -- Render input
+  -> m (InputElement EventResult (DomBuilderSpace m) t, DynValidation t Text a)
+manageValidation validate' validator renderInput = do
+  input <- renderInput
+  let currentVal = current $ value input
+  validatedInput <- fmap (fmap validator) $ buildDynamic (sample currentVal) $ tagPromptlyDyn (value input) validate'
+  return (input, toDynValidation validatedInput)
+
+validateEmail :: Text -> Either Text Text
+validateEmail m = do
```
Comment (3noch) [inline frontend/Rhyolite/Frontend/Form.hs]: Is there a combinator to set `type="email"` for inputs that use this validator?

Comment (3noch) [reply frontend/Rhyolite/Frontend/Form.hs]: Similar combinators for min/max/step/password/etc could exist to keep validation lined up with semantic input tags, used by mobile browsers to provide better UIs.

**Comment by ali-abrar** [comment]
@luigy @danbornside can you guys split this into two PRs: one for validation, one for routes?

**Comment by ryantrinkle** [comment]
@danbornside After splitting the PR, would you mind writing a couple sentences just explaining the motivation/approach of each?

**Pull request merged**